### PR TITLE
Adding GEFS forecast data source

### PR DESCRIPTION
**Pull request by NickGeneva**
# Earth2Studio Pull Request

## Description

Need to add unit tests.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/earth2studio/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/earth2studio/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/earth2studio/issues) is linked to this pull request.

### CODE REVIEW

File: test/data/test_gefs.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    ],
+)
+@pytest.mark.parametrize(
+    "lead_time",
+    [
+        timedelta(hours=0),
+        [
+            timedelta(hours=258),
+        ],
+    ],
```
Comment (dallasfoster) [inline test/data/test_gefs.py]: Do we need to test more than one time stamp?

File: test/data/test_gefs.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+)
+@pytest.mark.parametrize("variable", ["t2m", ["msl"]])
```
Comment (dallasfoster) [inline test/data/test_gefs.py]: Do we need to test more than one variable?

File: test/data/test_gefs.py
Comment (dallasfoster) [inline test/data/test_gefs.py:1]: Should we add a negative test for a wrong variable?

**Review by dallasfoster — APPROVED** [review]

File: test/data/test_gefs.py
Comment (NickGeneva) [reply test/data/test_gefs.py:1]: Added

File: test/data/test_gefs.py (hunk 2 above)
Comment (NickGeneva) [reply test/data/test_gefs.py]: Yep, added

**Pull request merged**